src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -1609,12 +1667,22 @@ int D4NFilterObject::D4NFilterDeleteOp::delete_obj(const DoutPrefixProvider* dpp
 
       if (!objDirty) { // object written to backend  
 	return next->delete_obj(dpp, y, flags);
+      } else {
+	if (!(ret = source->driver->get_policy_driver()->get_cache_policy()->eraseObj(dpp, policy_prefix, y))) {
+	  ldpp_dout(dpp, 0) << "Failed to delete policy object entry for: " << source->get_name() << ", ret=" << ret << dendl;
+	  return -ENOENT;
+        } else {
+          return 0;
+        }
       }
     } else { // provided version is a delete marker; remove delete marker block
       rgw::d4n::CacheBlock deleteBlock;
       deleteBlock = block;
       block.cacheObj.objName = source->get_name();
 
+      if (block.version == "null") 
```

Comment:
For some reason, I observed that requests with version-ids set to `null` have an oid that is equivalent to the object name, which is why I handle `null` version blocks this way.

Reply:
ok